rust/ql/lib/codeql/rust/controlflow/internal/CfgConsistency.qll

```diff
@@ -54,6 +56,24 @@ query predicate deadEnd(CfgImpl::Node node) {
   not letElsePanic(node.getAstNode())
 }
 
+pragma[nomagic]
+private predicate successfullyExtractedFile(File f) {
+  not exists(ExtractionWarning ee | ee.getLocation().getFile() = f)
+}
+
+query predicate missingCfgChild(CfgNode parent, string pred, int i, AstNode child) {
+  CfgNodes::missingCfgChild(parent, pred, i, child) and
+  successfullyExtractedFile(child.getLocation().getFile()) and
```

Comment:
Is there a reason why wee need this predicate for this consistency check specifically? For instance, I've seen dead ends in syntactically incorrect extracted Rust files. Should we instead do it for all consistency checks?

Reply:
We may want to do this for all (CFG) consistency checks.